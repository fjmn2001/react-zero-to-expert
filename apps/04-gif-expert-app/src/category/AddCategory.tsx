import {
  ChangeEvent,
  Dispatch,
  FormEvent,
  SetStateAction,
  useState,
} from "react"

const AddCategory = ({
  setCategories,
}: {
  setCategories: Dispatch<SetStateAction<string[]>>
}) => {
  const [inputValue, setInputValue] = useState("")

  const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
    setInputValue(e.currentTarget.value)
  }

  const handleSubmit = (e: FormEvent) => {
    e.preventDefault()

    if (inputValue.trim().length === 0) {
      return
    }

    setCategories((categories) => [inputValue, ...categories])
    setInputValue("")
  }

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  )
}

export default AddCategory
