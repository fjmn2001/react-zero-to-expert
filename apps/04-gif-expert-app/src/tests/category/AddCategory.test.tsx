import { describe, expect, test } from "vitest"
import { shallow } from "enzyme"
import AddCategory from "../../category/AddCategory"

describe("Testing AddCategory", () => {
  test("it should show component correctly", () => {
    const setCategories = () => {}
    const wrapper = shallow(<AddCategory setCategories={setCategories} />)
    expect(wrapper).toMatchSnapshot()
  })
})
