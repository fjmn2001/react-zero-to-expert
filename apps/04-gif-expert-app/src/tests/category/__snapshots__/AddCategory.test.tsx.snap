// Vitest Snapshot v1

exports[`Testing AddCategory > it should show component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
